File: groups/group5/Java-Recursion/Collatz/Collatz.java
package Collatz;

import java.util.ArrayList;

public class Collatz {
    public static String process(String[] inputs) throws Exception {
        int start = Integer.parseInt(inputs[0].trim());
        if (start <= 0 || start % 2 == 0)
            throw new Exception("Please provide a positive odd integer.");
        ArrayList<Integer> seq = new ArrayList<>();
        int current = start, peak = start;
        seq.add(current);
        while (current > 1) {
            current = (current % 2 == 0) ? current / 2 : (3 * current) + 1;
            if (current > peak)
                peak = current;
            seq.add(current);
        }
        StringBuilder sb = new StringBuilder("Collatz sequence for ").append(start).append(":\n");
        for (int i = 0; i < seq.size(); i++)
            sb.append(seq.get(i)).append(i == seq.size() - 1 ? "" : ", ");
        sb.append("\n\nTerms: ").append(seq.size()).append("\nSteps: ").append(seq.size() - 1).append("\nPeak value: ")
                .append(peak);
        return sb.toString();
    }
}

File: groups/group3/Collatz.js
function main(inputs) {
    const input = String(inputs[0] ?? "").trim();
    
    // Validation
    if (input === '') {
        throw new Error('Please enter a value');
    }
    
    if (input.includes('.')) {
        throw new Error('Decimals are not allowed. Please enter a whole number.');
    }
    
    const n = parseInt(input);
    
    if (isNaN(n)) {
        throw new Error('INVALID OUTPUT: Please enter a valid integer');
    }
    
    if (n <= 0) {
        throw new Error('INVALID OUTPUT: Number must be positive');
    }
    
    if (n % 2 === 0) {
        throw new Error('INVALID OUTPUT: Number must be odd');
    }
    
    // Generate Collatz sequence
    const sequence = [];
    let current = n;
    
    while (current !== 1) {
        sequence.push(current);
        if (current % 2 === 0) {
            current = Math.floor(current / 2);
        } else {
            current = 3 * current + 1;
        }
    }
    sequence.push(1);
    
    // Return formatted result
    return `✅ Collatz Sequence

Initial value: ${n}
Sequence: ${sequence.join(', ')}
Number of steps: ${sequence.length - 1}`;
}
